### DISCUSSION TITLE
Nextjs 14: Call server action from client component and listen for multiple promises

### DISCUSSION THREAD

**Discussion by Nils19024** [vercel/next.js#64067] [Help]
### Summary

I have a file database.jsx with server actions:

```
"use server"

export async function subscribeToRecord() {
   await pb.collection("user").subscribe(pb.authStore.model.id, function test(e) {
      console.log(e.action);
      return true
    })
}
```

and a client component file page.jsx that calls the server action function `subscribeToRecord `:

```
"use client"

import { subscribeToRecord } from "@/app/(serverActions)/database";

export default function Page() {

    subscribeToRecord().then(change => {
      console.log(change)
    });

   return ()
}
```

Currently i am not receiving any message in the client component when something is changing in the database. How can i achieve this? Can someone help me to find a solution for this problem? Thanks a lot!

**Comment by icyJoseph** [comment]
I think, with one promise this won't fly... but you could have a stream. 

This would also work with websockets, but server actions, resolve to a value. In general promises, once resolved, are moved to settle, and you can't change the resolved value.

I think the recommendation here is to use a Route Handler rather. And from there stream to your client maybe?

- https://nextjs.org/docs/app/building-your-application/routing/route-handlers#streaming look at the second example, iteratorToStream

**Reply by Nils19024** [reply]
Hi thanks for the answer. Can you give me a short example how i can do that with App Routing and streaming? I tried it, but did not came up with a solution 😅.
